packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      checkOperationResult(deleteResult1, { ...todo, ...todoUpdated1, privateContent: null, ownerContent: null }, deleteResultSetName);
+
+      // user2 can update the private field
+      const todoUpdated2 = {
+        customId: todo['customId'],
+        privateContent: 'Private Content updated 1',
+      };
+      const updateResult2 = await user2TodoHelper.update(updateResultSetName, todoUpdated2, privateResultSet);
+      expect(updateResult2.data[updateResultSetName].customId).toEqual(todo['customId']);
+      expectNullFields(updateResult2.data[updateResultSetName], ['privateContent']);
+      */
+
+      // user2 can read the allowed fields
+      const getResult2 = await user2TodoHelper.get(
+        {
+          customId: todo['customId'],
+        },
+        privateResultSet,
+        false,
+        'all',
+        'customId',
+      );
+      checkOperationResult(
+        getResult2,
+        {
+          customId: todo['customId'],
+          privateContent: todoUpdated1['privateContent'],
+        },
+        `get${modelName}`,
+      );
+
+      const listTodosResult2 = await user2TodoHelper.list({}, privateResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['customId'], true, 'customId');
+
+      // owner(user1) can listen to updates on allowed fields
+      const todoRandom = {
+        ...todo,
+        customId: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        customId: todoRandom.customId,
+        author: userName1,
+        privateContent: 'Private Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName1]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await user1TodoHelper.create(createResultSetName, todoRandom, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, privateContent: null, ownerContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await user1TodoHelper.update(updateResultSetName, todoRandomUpdated, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, privateContent: null, ownerContent: null },
+        `onUpdate${modelName}`,
+      );
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket